ext/mini_racer_extension/mini_racer_extension.cc

```diff
@@ -1691,6 +1698,14 @@ rb_heap_snapshot(VALUE self, VALUE file) {
     return Qtrue;
 }
 
+static VALUE
+rb_context_current_thread_clock(VALUE self) {
+	clockid_t thread_clock;
+	int ok = pthread_getcpuclockid(pthread_self(), &thread_clock);
```

Comment:
We might actually be able to (since we are in ruby in this function), just call a ruby function for getting clockid and let it handle the multiplatform bit.

Reply:
> Dangit! Might not be supported on macos- [nanomsg/nanomsg#10](https://github.com/nanomsg/nanomsg/issues/10)

That's from 2013 and about `CLOCK_MONOTONIC` not `pthread_getcpuclockid`. POSIX 1.2017 specifies `CLOCK_MONOTIC` in Base and `pthread_getcpuclockid` with the `[TCT]` optional code so I'd be surprised if at least `CLOCK_MONOTONIC` wasn't on macOS by now. CPU clock times might not be, in which case we'd need some fallback.

> We might actually be able to (since we are in ruby in this function), just call a ruby function for getting clockid and let it handle the multiplatform bit.

I don't believe Ruby's threading system support for getting a native clock id -- you don't even get native *thread* ID until, like, Ruby 3.1; and even then it's the *wrong* thread id (i.e. not `pthread_t`)

Reply:
Slight correction: `CLOCK_MONOTONIC` is `[MON]`-optional in POSIX.1-2017 but in Base since POSIX.1-2024... that's not a great sign

Reply:
Asked around on my end; at least macOS 10.15 has `CLOCK_MONOTONIC`, but no `pthread_getcpuclockid`... but it *does* have `CLOCK_THREAD_CPUTIME_ID`, despite those being set in the same option?? At least it doesn't define `_POSIX_THREAD_CPUTIME` either, so we can condition things on that, if we need to.

Reply:
More bad news and good news:
https://github.com/comcode-org/mini_racer/pull/10#issuecomment-2400353780

Going to try and set up a codespace for dev on this.